Mu2eG4/src/constructCRV.cc

```diff
@@ -76,23 +83,216 @@ namespace mu2e
 
       if(verbosityLevel > 0) cout << __func__ << " constructing            : " << CRVsectorName << endl;
 
-      // all materials and dimensions are the same within a particular CRV sector
-      CRSScintillatorBarDetail const & barDetail = shield.getCRSScintillatorBarDetail();
+      /*******************************************/
+      /**** mother volumes around CRV sectors ****/
+      /*******************************************/
+      const std::vector<CRSScintillatorModule> &modules = shield.getCRSScintillatorModules();
+      const CRSScintillatorModule &module0 = modules.front(); //1st module of CRV sector
+      const CRSScintillatorModule &module1 = modules.back();  //last module of CRV sector
+
+      int thicknessDirection=shield.getCRSScintillatorBarDetail().getThicknessDirection();
+
+      //determine the dimensions of all combined layers of a sector
+      //e.g. the 3rd layers of all modules are combined into one big layer
+      //(ordered from the strong back to the thin aluminum cover sheet)
+      std::vector<CLHEP::Hep3Vector> motherCenterInMu2e;
+      std::vector<std::vector<double> > motherHalfLengths;
+      //collect all layers
+      for(size_t l=0; l<module0.getLayers().size(); ++l)
+      {
+        //strong back dimensions
+        if(l==0)
+        {
+          const CRSAluminumSheet &aluminumSheet0 = module0.getAluminumSheets().at(0);
+          const CRSAluminumSheet &aluminumSheet1 = module1.getAluminumSheets().at(0);
+          const CLHEP::Hep3Vector &centerInMu2e0=aluminumSheet0.getPosition();
+          const CLHEP::Hep3Vector &centerInMu2e1=aluminumSheet1.getPosition();
+          motherCenterInMu2e.emplace_back(0.5*(centerInMu2e0+centerInMu2e1));
+          std::vector<double> halfLengths;
```

Comment:
I believe these (and similarly used) halfLengths could be just C-style arrays and could be declared outside the loops they are in (and if the vectors or std::arrays were to be used std::vector<double> halfLengths(3) would be better)